src/iris_grib/tests/integration/load_convert/test_sample_file_loads.py

```diff
@@ -197,5 +219,66 @@ def test_shape_of_earth_grib1(self):
         self.assertCML(cube, _RESULTDIR_PREFIX + ("earth_shape_grib1.cml",))
 
 
+class TestTimesGrib1:
+    # Our codebase has support for many timeRangeIndicator values in GRIB1
+    #  files, but we do not have files demonstrating these. This class
+    #  generates appropriate GRIB1 files so that we can fully refactor the
+    #  loading code while ensuring continued support for all possible
+    #  scenarios.
+
+    @pytest.fixture(params=[1, 2, 3, 4, 5, 10, 113, 118, 123, 124], autouse=True)
+    # Note that the following values are not supported by Eccodes - it can
+    #  not provide a startStep value unless the timeRangeIndicator is present
+    #  in eccodes/definitions/grib1/localConcepts/edzw/stepType.def:
+    #  [51, 114, 115, 116, 117, 125].
+    def _get_time_range_file(self, request, tmp_path):
+        save_file = tmp_path / "TestTimes.grib1"
+        time_range_indicator = request.param
+
+        # Make a file with 10 ascending time steps.
+        with save_file.open("wb") as open_file:
+            for time_step in range(10):
+                grib_message = eccodes.codes_grib_new_from_samples("GRIB1")
```

Comment:
How confident are we about the stability of this? It hasn't changed since 2006 (we think), but it would be possible to cache a 'vanilla' file in iris-test-data if desired.

Reply:
I don't think we need to worry about this.
The GRIB2 code uses the same strategy, and it hasn't proven a big problem (it is detail changes in handling of particular keys that seem to upset us)